src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,321 @@
+                        destroyTooltipOnHide={false}
+                        onPopupAlign={placeArrowBottomLeft}
+                    >
+                        {this.mainContent(`No alterations in ${this.getHugoSymbolName(clinicalGroupMatch.notMatches, 3)} defined by the trial`)}
+                    </DefaultTooltip>
+                }
+            </span>
+        );
+    }
+
+    public tooltipGenomicContent(data: Array<IGenomicGroupMatch>) {
+        return (
+            <div className={styles.tooltip}>
+                {data.map((genomicGroupMatch: IGenomicGroupMatch) => (
+                    <div className={styles.genomicSpan}><b>Not </b>{`${genomicGroupMatch.genomicAlteration.replace(/!/g, '')} `}
+                        {genomicGroupMatch.matches[0].sampleIds.map((sampleId: string) => (
+                            <span className={styles.genomicSpan}>
+                                {this.props.sampleManager!.getComponentForSample(sampleId, 1, '')}
+                            </span>
+                        ))}
+                    </div>
+                ))}
+            </div>
+        );
+    }
+    public tooltipClinicalContent(data: Array<string>) {
+        return (
+            <div className={styles.tooltip}>
+                <ul className={styles.alterationUl}>
+                    {data.map((cancerType: string) => (
+                        <li>{cancerType}</li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+
+    public mainContent(content: string) {
+        return (
+            <span>
+                <a>{content}</a>
+            </span>
+        );
+    }
+
+    public getHugoSymbolName(matches: Array<IGenomicGroupMatch>, threshold: number) {
+        const hugoSymbolSet = new Set([...matches].map(x => x.genomicAlteration.split(' ')[0].slice(1)));
+        if (hugoSymbolSet.size <= threshold) {
+            return [...hugoSymbolSet].join(', ');
+        }
+        return `${hugoSymbolSet.size} genes`;
+    }
+
+    public buildDetailedTrialMatches() {
+        const groupByTrialId = _.groupBy(this.props.trialMatches, 'nctId');
+        let matchedTrials: Array<IDetailedTrialMatch> = [];
+        _.forEach(groupByTrialId, (trialGroup, trialId) => {
+            let matchedTrial:any = _.find( this.props.trials, { 'nctId': trialId } );
+            matchedTrial['priority'] = 0; // highest priority
+            matchedTrial['matches'] = [];
+            const groupByArm = _.groupBy(trialGroup, 'armDescription');
+            _.forEach(groupByArm, (armGroup, armDescription) => {
+                let armMatch: IArmMatch = {
+                    armDescription: armDescription,
+                    drugs: [],
+                    matches: []
+                };
+                const isControlArm = getDrugsAndExcludeControlArms(armDescription, matchedTrial['treatmentList']['step'][0]['arm'], armMatch);
```

Comment:
instead of matchedTrial['treatmentList'] (which i don't think is typesafe), use dot notation matchedTrial.treatmentList so typescript will check it

Reply:
Changed to 
```
const isControlArm = getDrugsAndExcludeControlArms(armDescription, originalMatchedTrial.treatmentList.step[0].arm, armMatch);
```